File: WebApplicationRefregitz/TakeRoot.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;

namespace Refrigtz
{
    [Serializable]
    public static class TakeRoot
    {
        static AllDraw Node = null;
        //  POINTER IS THE MEMMERY LOCATION OF LAST MOVMENTS.
        public static AllDraw Pointer=null;
        
        public static void CalculateRootGray(AllDraw Curent)
        {
            try
            {
                if (Node != null)
                {
                    if (Node.Dept != null)
                        Pointer.Dept = Curent;
                    else
                        Node.Dept = Curent;
                }
                else
                    Node = Curent;
            }
            catch (Exception tt)
            {
                return;
            }
            return;
        }
       
    }
}

File: Refrigtz/TakeRoot.vb
Imports System.Text
Imports System.Linq
Imports System.Drawing
Imports System.Collections.Generic
Imports System

Namespace Refrigtz
	<Serializable> _
	Public Shared Class TakeRoot
		Shared Node As AllDraw = Nothing
		'  POINTER IS THE MEMMERY LOCATION OF LAST MOVMENTS.
		Public Shared Pointer As AllDraw = Nothing

		Public Shared Sub CalculateRootGray(Curent As AllDraw)
			Try
				If Node IsNot Nothing Then
					If Node.Dept IsNot Nothing Then
						Pointer.Dept = Curent
					Else
						Node.Dept = Curent
					End If
				Else
					Node = Curent
				End If
			Catch tt As Exception
				Return
			End Try
			Return
		End Sub

	End Class
End Namespace